snuba/consumers/consumer_builder.py

```diff
@@ -55,19 +55,21 @@ def __init__(
             self.bootstrap_servers = bootstrap_servers
 
         stream_loader = enforce_table_writer(self.dataset).get_stream_loader()
-        self.raw_topic = raw_topic or stream_loader.get_default_topic_spec().topic_name
-        default_replacement_topic_name = (
-            stream_loader.get_replacement_topic_spec().topic_name
-            if stream_loader.get_replacement_topic_spec()
-            else None
+        self.raw_topic = (
+            Topic(raw_topic)
```

Comment:
Using Topic here when there is a KafkaTopic makes me think that maybe the naming is wrong. It seems the difference between Topic and KafkaTopic is not that one is abstract and the other is Kafka specific, but that one is a dataclass that identifies a Kafka topic and provides the parameters to create it and the other is just identifying a Topic.

What about preserving the TopicSpec name or something like TopicTemplate ? So you would preserve the ability to pass that class around instead of extracting a string and rebuilding it all the times but still we would not introduce a confusing naming.

Reply:
> It seems the difference between Topic and KafkaTopic is not that one is abstract and the other is Kafka specific, but that one is a dataclass that identifies a Kafka topic and provides the parameters to create it and the other is just identifying a Topic.

Yeah, that is a good way of describing it. 

> What about preserving the TopicSpec name or something like TopicTemplate ? So you would preserve the ability to pass that class around instead of extracting a string and rebuilding it all the times but still we would not introduce a confusing naming.

I have mixed feelings about this. I'll give it some more thought, but here is what I would be looking for in the ideal name:

- is obvious that it is substitutable with the `Topic` superclass (a rule of thumb here for me is that `Topic` should be the last word in the name)
- suggests at least loose coupling with the Kafka backend
- includes some notion that the object includes configuration parameters

Maybe it would make sense to take this a different direction, and make the configuration class wrap the `Topic` instance, like this — sort of a middle ground between the previous implementation and this one:

```python
class KafkaTopicSpecification:
    topic: Topic
    partitions_number: int
    … etc …
```

This would lose substitutability (which maybe is an abuse of inheritance in this case anyway), but would still have the advantage of not having to repeatedly transform strings with `Topic` instances. It maintains the last two points above.